sdk/serialization/azure-json/src/main/java/com/azure/json/JsonWriter.java

```diff
@@ -546,14 +715,45 @@ public final <T> JsonWriter writeArrayField(String fieldName, T[] array,
      */
     public final <T> JsonWriter writeArrayField(String fieldName, Iterable<T> array,
         WriteValueCallback<JsonWriter, T> elementWriterFunc) throws IOException {
+        return writeArrayField(fieldName, array, elementWriterFunc, false);
+    }
+
+    /**
+     * Writes a JSON array field.
+     * <p>
+     * This API will begin by writing the field name and start array ({@code [}) followed by all elements in the array
+     * using the {@code elementWriterFunc} and finishing by writing the end array ({@code ]}).
+     * <p>
+     * If an element in the array is null, whether writing {@link JsonToken#NULL} for the element will be determined by
+     * the {@code elementWriterFunc}'s handling of null values. For example, {@link #writeJson(JsonSerializable)} will
+     * skip writing null values, so if that is used nothing will be written for a null element.
```

Comment:
This is only true is `explicitlyWriteNull` is false. If this is true, regardless of what the `elementWriterFunc` does, we will write `null` value.

Reply:
Changing design